homeassistant/components/home_connect/__init__.py

```diff
@@ -158,16 +178,37 @@ async def _async_service_program(call, method):
                 option[ATTR_UNIT] = option_unit
 
             options.append(option)
-
-        appliance = _get_appliance(hass, device_id)
-        await hass.async_add_executor_job(getattr(appliance, method), program, options)
+        appliance = _get_appliance_or_rise_service_validation_error(device_id, hass)
+        try:
+            await hass.async_add_executor_job(
+                getattr(appliance, method), program, options
+            )
+        except api.HomeConnectError as err:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key=method,
+                translation_placeholders={
+                    **get_dict_from_home_connect_error(err),
+                    SVE_TRANSLATION_PLACEHOLDER_PROGRAM: program,
+                },
+            ) from err
```

Comment:
Maybe this could be generic also:


```python
async def _run_appliance_service(hass: HomeAssistant, appliance: X, method: str, args: tuple[X], error_code: str, error_placeholders: dict[str, str]) -> None:
        try:
            await hass.async_add_executor_job(
                getattr(appliance, method), *args
            )
        except api.HomeConnectError as err:
            raise ServiceValidationError(
                translation_domain=DOMAIN,
                translation_key=error_code,
                translation_placeholders={
                    **get_dict_from_home_connect_error(err),
                    **error_placeholders,
                },
            ) from err

...
await _run_appliance_service(
    hass,
    _get_appliance_or_raise_service_validation_error(device_id, hass),
    method,
    (program, options),
    method,
    {"program": program}
)
```